libraries/Microsoft.Bot.PublishValidation/Helpers/BotValidatorHelper.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Bot.PublishValidation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using Microsoft.Bot.Configuration;
+
+    public class BotValidatorHelper
+    {
+        private const string FORBIDDEN_ENDPOINTS = "Forbidden Endpoints";
+        private const string REQUIRED_ENDPOINTS = "Required Endpoints";
+
+        /// <summary>
+        /// Checks if a bot file is valid or not according to some configuration options
+        /// </summary>
+        /// <param name="configurationOptions"></param>
+        /// <param name="messages"></param>
+        /// <returns></returns>
+        public static bool BotFileIsValid(ConfigurationOptions configurationOptions, ref IEnumerable<NotificationMessage> messages)
+        {
+            try
+            {
+                // Load the first .bot file from the provided folder.
+                // Also validates its existence (throws an exception is there isn't any file)
+                var botConfiguration = BotValidatorHelper.LoadFromFolder(configurationOptions.ProjectPath);
+
+                return BotValidatorHelper.ValidateBotFile(botConfiguration, configurationOptions, ref messages);
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
```

Comment:
Similar comment. Not sure why there is a try/catch here. If we want to keep it (handy for breakpoints) then change the "throw ex" to just "throw" so the exception isn't munched.